src/itemstackmetadata.cpp

```diff
@@ -0,0 +1,43 @@
+#include "itemstackmetadata.h"
+#include "util/serialize.h"
+#include "util/strfnd.h"
+
+#define DESERIALIZE_START '\x01'
+#define DESERIALIZE_KV_DELIM '\x02'
+#define DESERIALIZE_PAIR_DELIM '\x03'
+#define DESERIALIZE_START_STR "\x01"
+#define DESERIALIZE_KV_DELIM_STR "\x02"
+#define DESERIALIZE_PAIR_DELIM_STR "\x03"
+
+void ItemStackMetadata::serialize(std::ostream &os) const
+{
+	std::ostringstream os2;
+	os2 << DESERIALIZE_START;
+	for (StringMap::const_iterator
+			it = m_stringvars.begin();
+			it != m_stringvars.end(); ++it) {
+		os2 << it->first << DESERIALIZE_KV_DELIM
+		   << it->second << DESERIALIZE_PAIR_DELIM;
+	}
+	os << serializeJsonStringIfNeeded(os2.str());
+}
+
+void ItemStackMetadata::deSerialize(std::istream &is)
+{
+	std::string in = deSerializeJsonStringIfNeeded(is);
```

Comment:
string copy, please use a reference or a pointer as parameter of deSerialize function for your result